processing/src/main/java/org/apache/druid/segment/NestedDataColumnIndexer.java

```diff
@@ -142,29 +142,26 @@ public DimensionSelector makeDimensionSelector(
       IncrementalIndex.DimensionDesc desc
   )
   {
-    if (
-        fieldIndexers.size() == 1 &&
-        fieldIndexers.containsKey(NestedPathFinder.JSON_PATH_ROOT) &&
-        fieldIndexers.get(NestedPathFinder.JSON_PATH_ROOT).getTypes().getSingleType() != null
-    ) {
-      final ColumnValueSelector delegate = makeColumnValueSelector(currEntry, desc);
+    final int dimIndex = desc.getIndex();
+    final ColumnValueSelector<?> rootLiteralSelector = getRootLiteralValueSelector(currEntry, dimIndex);
+    if (rootLiteralSelector != null) {
       return new BaseSingleValueDimensionSelector()
       {
         @Nullable
         @Override
         protected String getValue()
         {
-          final Object o = delegate.getObject();
+          final Object o = rootLiteralSelector.getObject();
           if (o == null) {
             return null;
           }
-          return String.valueOf(o);
+          return o.toString();
         }
 
         @Override
         public void inspectRuntimeShape(RuntimeShapeInspector inspector)
         {
-          delegate.inspectRuntimeShape(inspector);
+
```

Comment:
I'll be honest that I'm unclear on what `inspectRuntimeShape` actually does, but you removed the call to the delegate instead of changing it to call on `rootLiteralSelector`.  Is that intentional?

Reply:
`inspectRuntimeShape` is used with the topN specialization stuff which doesn't do much if anything to incremental index queries afaik. i removed it from here because `rootLiteralSelector` method also does nothing so it seemed pointless